src/signals/settings/users/containers/Detail/hooks/useFetchUser.js

```diff
@@ -1,59 +1,104 @@
 import { useState, useEffect } from 'react';
-
-import CONFIGURATION from 'shared/services/configuration/configuration';
 import { getAuthHeaders } from 'shared/services/auth/auth';
-
-export const userEndpoint = `${CONFIGURATION.API_ROOT}signals/v1/private/users/`;
+import { getErrorMessage } from 'shared/services/api/api';
+import configuration from 'shared/services/configuration/configuration';
 
 /**
  * Custom hook useFetchUser
  *
- * Will call private /users endpoint
+ * Will call private /users endpoint for both GET and PATCH
  *
  * @returns {FetchResponse}
  */
 const useFetchUser = id => {
   const [isLoading, setLoading] = useState();
-  const [data, setData] = useState();
+  const [isSuccess, setSuccess] = useState();
+  const initialState = id ? undefined : {};
```

Comment:
why do we need this?

Reply:
Initial state is set to `undefined` when `id` has a value, which means that the form is loading with the intention to edit existing user data. Since the data still has to be retrieved, I don't want to show the form until the data is available.
When `id` is absent, the data available to the `UserDetail` component is an empty object so that it renders the form with empty fields.